### ISSUE TITLE
Error: '<variable>' is assigned to itself

### ISSUE THREAD

**Issue by probablykasper** [sveltejs/eslint-plugin-svelte#454]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.

### What version of ESLint are you using?

8.39.0

### What version of `eslint-plugin-svelte` are you using?

2.27.1

### What did you do?

<details>
<summary>Configuration</summary>

```js
module.exports = {
	root: true,
	extends: [
		'eslint:recommended',
		'plugin:@typescript-eslint/recommended',
		'plugin:svelte/recommended',
		'prettier'
	],
	parser: '@typescript-eslint/parser',
	plugins: ['@typescript-eslint'],
	ignorePatterns: ['*.cjs'],
	parserOptions: {
		sourceType: 'module',
		ecmaVersion: 2020,
		extraFileExtensions: ['.svelte']
	},
	env: {
		browser: true,
		es2017: true,
		node: true
	},
	overrides: [
		{
			files: ['*.svelte'],
			parser: 'svelte-eslint-parser',
			parserOptions: {
				parser: '@typescript-eslint/parser'
			}
		}
	]
};

```
</details>

1. `npm create svelte@latest`
2. Put the following in `src/routes/+page.svelte`

```svelte
<script lang="ts">
	let x = new Date();
	setTimeout(() => {
		x.setFullYear(2020);
		x = x;
	}, 1000);
</script>
```

### What did you expect to happen?

Expected no error because self-assignments are used in Svelte to trigger updates

### What actually happened?

ESLint emits this error when using the default config provided with `npm create svelte@latest`:
```
  5:9  error  'x' is assigned to itself  no-self-assign
```

### Link to **GitHub Repo** with Minimal Reproducible Example

https://github.com/probablykasper/eslint-svelte-self-assign-bug

**Comment by ota-meshi** [comment]
thanks for letting me know.
